pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -124,19 +126,52 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 
 	sts, err := buildStatefulSet(mdb)
 	if err != nil {
-		log.Warnf("error building StatefulSet: %s", err)
+		log.Infof("Error building StatefulSet: %s", err)
 		return reconcile.Result{}, nil
 	}
 
 	if err = r.client.CreateOrUpdate(&sts); err != nil {
-		log.Warnf("error creating/updating StatefulSet: %s", err)
+		log.Infof("Error creating/updating StatefulSet: %s", err)
 		return reconcile.Result{}, err
+	} else {
+		log.Infof("StatefulSet successfully Created/Updated")
+	}
+
+	log.Infof("waiting for StatefulSet %s/%s to reach ready state", mdb.Namespace, mdb.Name)
+	set := appsv1.StatefulSet{}
+	timedOut, err := r.client.WaitForCondition(types.NamespacedName{Name: mdb.Name, Namespace: mdb.Namespace}, time.Second*3, time.Second*30, &set, func() bool {
+		return statefulset.IsReady(set)
+	})
+
+	if timedOut {
```

Comment:
This kind of information is only relevant in a debugging scenario. Moreover, I think we can go with a different strategy here. We should rely on reconcile loop for the mechanics of object polling & waiting, not on our own loops with sleeps in between.

Reply:
The idea here was a relatively short wait period before retrying. We could of course shorten it to almost no wait period, this would end up with a lot of reconciliations consistently requeued (maybe that's not a problem though!)

Reply:
In the Kubernetes environment, a blocking controller is a problem.